### PR TITLE
Tag backup and restore tool

### PR DESCRIPTION
**Pull request by jamesbrown7**

### CODE REVIEW

File: examples/BackupRestoreTags/tagutil.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
+package main
```
Comment (ppg) [inline examples/BackupRestoreTags/tagutil.go]: Should we just call this file `main.go`? The `BackupRestoreTags` folder seems to name the tool?

**Comment by ppg** [comment]
:+1:

File: examples/BackupRestoreTags/tagutil.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+}
+
+func findTagByName(tags []jcapi.JCTag, tagName string) (idx int) {
+	for idx, value := range tags {
+		if value.Name == tagName {
+			return idx
+		}
+	}
+
+	return -1
```
Comment (rmulley) [inline examples/BackupRestoreTags/tagutil.go]: Might be more Go-ish to return an error in this case instead.

Comment (jamesbrown7) [inline examples/BackupRestoreTags/tagutil.go]: I wouldn't consider it to be an error to not find an object I was looking for. This is common behavior for search functions in core Go packages, like https://golang.org/pkg/strings/#Index

Comment (rmulley) [inline examples/BackupRestoreTags/tagutil.go]: I'm fine with that.  I was thinking of how `database/sql` returns an error for no rows.

**Comment by rmulley** [comment]
:+1:

**Pull request merged**